File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(pr, location) {
    var answer = 0;
    while(pr.length !== 0){
        let poppedPr = pr[0];
        if(poppedPr === Math.max(...pr)){
            pr.splice(0,1);
            answer += 1;
            if(location === 0) break;
        }
        else{
            pr.splice(0,1);
            pr.push(poppedPr);
        }
        location -= 1;
        if(location < 0){
            location += pr.length;
        }
    }
    return answer;
}

File: 프로그래머스/2/42587. 프로세스/프로세스.cpp
#include <bits/stdc++.h>

using namespace std;

priority_queue<int> pq;
queue<pair<int,int>> q;

int solution(vector<int> priorities, int location) {
    int answer = 0;
    int size = priorities.size();
    
    for(int i=0;i<size;i++){
        pq.push(priorities[i]);
        q.push({priorities[i],i}); // 우선순위, 인덱스
    }
    
    while(!q.empty()){
        int priority = q.front().first;
        int index = q.front().second;
        q.pop();
        
        if(pq.top() == priority){
            pq.pop();
            answer++;
            if(index == location) return answer;
        }
        else{
            q.push({priority,index});
        }
    }
    
    return answer;
}